### DISCUSSION TITLE
SEO optimizable routes with dynamic routing

### DISCUSSION THREAD

**Discussion by BramDecuypere** [vercel/next.js#11171] [Help] [answered]
# Feature request

## Is your feature request related to a problem? Please describe.

What I'm trying to accomplish is with dynamic routing have a substructure as so:

Full path - for each subroute we would need a seperate page (specific page component as well)
`/[lang]/[overview-slug]/[brand]/[model]/[id]`
- Note that overview-slug is language specific

`/nl/overzicht`
`/nl/overzicht/audi`
`/nl/overzicht/audi/a3`
`/nl/overzicht/audi/a3/12430213-123-123123-123`

## Describe the solution you'd like

previous question

## Describe alternatives you've considered

Used to work with `next-routes`, but currently unable to export

## Additional context
Basic repository for reproduction: 
https://github.com/lizy-lab/lizy-basic-project

**Comment by timneutkens** [comment, marked as answer]
No question is asked 🤔 What's outlined works totally fine with dynamic routes.

**Reply by BramDecuypere** [reply]
@timneutkens Sorry for waisting your time. I failed to comprehend, (with my limited brainpower), the way the files should be defined for the subroutes to work. That the endpage had to have the same `[dynamic]` definition for both the folder and the file. So for anyone else having this problem, that is the solution. 

I'm amazed sometimes by how elegantly things are solved here at Next. Thanks for that!

```
/nl/overzicht
/nl/overzicht/audi
/nl/overzicht/audi/a3
/nl/overzicht/audi/a3/12430213-123-123123-123
```

**Comment by BramDecuypere** [comment]
I post it here since it's related

@timneutkens how would one go about conflict routes in this scenario? 

Or would you do this on page level to check the route and show a different page-component?

```
/[lang]/[overview]/[brand]
/[lang]/[about-us] --> translated routes that throw this error
```

throw new Error(`You cannot use different slug names for the same dynamic path ('${previousSlug}' !== '${nextSlug}').`);
